src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -86,44 +134,31 @@ class HttpGoogleCloudStorageDAO(
-    bucketDeleteFn()
-  }
 
-  override def createACLGroups(userInfo: UserInfo, workspaceId: String, workspaceName: WorkspaceName): Unit = {
-    val bucketName = getBucketName(workspaceId)
-    val groupDirectory = getGroupDirectory
-    val groups = groupDirectory.groups
-    val bucketAccessControls = getStorage(getBucketCredential(userInfo)).bucketAccessControls
-
-    // ACLs are stored internally in three Google groups.
-    groupAccessLevelsAscending foreach { accessLevel =>
-      // create the group
-      val groupId = toGroupId(bucketName, accessLevel)
-      val group = new Group().setEmail(groupId).setName(toGroupName(workspaceName,accessLevel))
-      val inserter = groups.insert(group)
-      retry(inserter.execute,when500)
-
-      // add group to ACL (using bucket owner's credential)
-      val bucketAcl = new BucketAccessControl().setEntity(makeGroupEntityString(groupId)).setRole(WorkspaceAccessLevel.toCanonicalString(accessLevel)).setBucket(bucketName)
-      val aclInserter = bucketAccessControls.insert(bucketName, bucketAcl)
-      retry(aclInserter.execute,when500)
+    val groups = getGroupDirectory.groups
+    deleteGroup(groups,toGroupId(bucketName,WorkspaceAccessLevel.Read)) {
+      deleteGroup(groups,toGroupId(bucketName,WorkspaceAccessLevel.Write)) {
+        deleteGroup(groups,toGroupId(bucketName,WorkspaceAccessLevel.Owner)) {
+          // attempt to delete the group, scheduling future attempts if this one fails due to a 409
+          Try(bucketDeleteFn()) match {
+            case Success(_) =>
+            case Failure(exc) => throw new RawlsException("Failed to delete bucket ${bucketName}.", exc)
+          }
+        }
+      }
     }
-
-    // add the owner -- may take up to 1 minute after groups are created
-    // (see https://developers.google.com/admin-sdk/directory/v1/guides/manage-groups)
-    val targetMember = new Member().setEmail(userInfo.userEmail).setRole(groupMemberRole)
-    val inserter = groupDirectory.members.insert(toGroupId(bucketName, WorkspaceAccessLevel.Owner), targetMember)
-    retry(inserter.execute,when500)
   }
 
-  override def deleteACLGroups(workspaceId: String): Unit = {
-    val groups = getGroupDirectory.groups
-    groupAccessLevelsAscending foreach { accessLevel =>
-      val groupId = toGroupId(getBucketName(workspaceId), accessLevel)
-      val deleter = groups.delete(groupId)
-      retry(deleter.execute,when500)
+  private def deleteGroup( groups: Directory#Groups, groupId: String)( op: => Unit ) = {
+    val deleter = groups.delete(groupId)
+    tryRetry(deleter.execute,when500) match {
+      case Success(_) => op
+      case Failure(exception) =>
+        val thisException = new RawlsException(s"Failed to delete group ${groupId}.", exception)
+        Try(op) match {
+          case Success(_) => throw thisException
+          case Failure(thatException) => throw new RawlsException(""+thisException.getMessage+"\nand\n"+thatException.getMessage)
```

Comment:
this exception chaining loses the important details of the originating exceptions, namely the stack. I think the first exception that gets thrown is the most important one to diagnose the issue